src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -582,12 +582,13 @@ private void InitNativeProcess()
                     }
                     catch (Win32Exception)
                     {
+#if UNIX
                         // On Unix platforms, nothing can be further done, so just throw
                         // On headless Windows SKUs, there is no shell to fall back to, so just throw
                         if (!Platform.IsWindowsDesktop) { throw; }
```

Comment:
Per the error, this shouldn't be wrapped in `#if UNIX` because it's supposed to fail on Windows systems that don't have a desktop (shell)

Reply:
Fixed.